Shared/Samples/Edit with branch versioning/EditWithBranchVersioningView.Views.swift

```diff
@@ -0,0 +1,152 @@
+                }
+                .navigationTitle("Create Version")
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .cancellationAction) {
+                        Button("Cancel") {
+                            model.clearSelection()
+                            dismiss()
+                        }
+                    }
+                    
+                    ToolbarItem(placement: .confirmationAction) {
+                        Button("Done") {
+                            let parameters = ServiceVersionParameters()
+                            parameters.access = selectedVersionAccess!
+                            parameters.name = versionName
+                            parameters.description = versionDescription
+                            
+                            action(parameters)
+                            dismiss()
+                        }
+                        .disabled(!inputsAreValid)
+                    }
+                }
+            }
+        }
+        
+        /// Validates a given name to ensure it meets the criteria for a version name.
+        /// - Parameter name: The name to validate.
+        /// - Returns: Text describing the broken validation rule, if any.
+        private func validateVersionName(_ name: String) -> String? {
+            if name.isEmpty {
+                return "Name is required"
+            } else if name.count > 62 {
+                return "Name must not exceed 62 characters"
+            } else if name.first == " " {
+                return "Name must not begin with a space."
+            } else if name.contains(";") {
+                return "Name must not contain semicolons (;)"
+            } else if name.contains(".") {
+                return "Name must not contain periods (.)"
+            } else if name.contains("'") {
+                return "Name must not contain single quotation marks (')"
+            } else if name.contains("\"") {
+                return "Name must not contain double quotation marks (\")"
+            } else if model.existingVersionNames.contains(name) {
+                return "Version already exists"
+            } else {
+                return nil
+            }
+        }
+    }
+}
+
+private extension VersionAccess {
+    /// All the version access cases.
+    static var allCases: [Self] { [.public, .protected, .private] }
+    
+    /// The text description of the version access.
+    var description: String {
+        switch self {
+        case .public: return "Any portal user can view and edit the version."
+        case .protected: return "Any portal user can view the version, but only the version owner, feature layer owner, and portal administrator can edit it."
+        case .private: return "Only the version owner, feature layer owner, and portal administrator can view and edit the version."
+        @unknown default: return "Unknown version access."
```

Comment:
Return keywords aren't needed:

```suggestion
        case .public: "Any portal user can view and edit the version."
        case .protected: "Any portal user can view the version, but only the version owner, feature layer owner, and portal administrator can edit it."
        case .private: "Only the version owner, feature layer owner, and portal administrator can view and edit the version."
        @unknown default: "Unknown version access."
```